pigeon/client.py

```diff
@@ -62,6 +66,19 @@ def __init__(
         )
         self._logger = logger if logger is not None else self._configure_logging()
 
+        self.pid = os.getpid()
+        self.process_name = psutil.Process(self.pid).name()
+        self.hostname = socket.gethostname().split('.')[0]
+        self.name = f'{self.hostname}_{self.pid}_{self.process_name}'
+
+    def announce(self, connected=True):
+        self.send("announce_connection", name=self.name, pid=self.pid, hostname=self.hostname,
+                  process_name=self.process_name, connected=connected)
+
+    def update_state(self):
+        self.send("update_state", name=self.name, pid=self.pid, hostname=self.hostname,
+                  process_name=self.process_name, subscribed_to=list(self._callbacks.keys()))
```

Comment:
Should `announce` and `update_state` be `_announce` and `_update_state`?

Reply:
That makes sense.